tools/monorepo-utils/src/md-docs/lib/generate-manifest.ts

```diff
@@ -24,30 +21,30 @@ function generatePageId( filePath: string, prefix = '' ) {
 	return hash.digest( 'hex' );
 }
 
-function generateHashOfString( str: string ) {
-	return crypto.createHash( 'sha256' ).update( str ).digest( 'hex' );
-}
-
-function generateRawGithubFileUrl(
-	owner: string,
-	repo: string,
-	gitBranch: string,
-	repoPath: string,
+export const generateFileUrl = (
+	baseUrl: string,
+	rootDirectory: string,
+	directory: string,
 	filePath: string
```

Comment:
I needed to read the tests to figure out what each variable did. I think a jsDoc would be helpful here.

Aside: At first I would think `absoluteFilePath` below to be the exact same as the `filePath` argument, but if I'm thinking about this correctly, a relative path would also be accepted? If thats correct, another test would be appropriate.

Reply:
Yes you're right this whole thing was a mess and the params unclearly named. 

This function actually expects all paths to be absolute for it to work. it determines from a combination of root path and sub directory and absolute file path how to generate a url.

I've added some error throwing if you pass a relative path and updated the tests to cover those cases too. I also realized the order of params / naming of params in other functions was a mess too so I've reordered them or renamed them for clarity where it made sense: https://github.com/woocommerce/woocommerce/pull/39161/commits/3ed77ef2d967dc61863d79c00bce1c8cadb102d3

Also I noticed in the process of doing this that I'd accidentally left off generating of the hash the manifest should have so I added that back and have added a brief test for that too.